presto-native-execution/presto_cpp/main/runtime-metrics/tests/PrometheusReporterTest.cpp

```diff
@@ -20,7 +20,7 @@ namespace facebook::presto::prometheus {
 class PrometheusReporterTest : public testing::Test {
  public:
   void SetUp() override {
-    reporter = std::make_shared<PrometheusStatsReporter>(testLabels);
+    reporter = std::make_shared<PrometheusStatsReporter>(testLabels, 1);
```

Comment:
Why not test with multiple threads?

Reply:
The problem is that for gauge metric its the last value which stays so the testing has to be in a single thread for it to be deterministic

Reply:
@jaystarshot : Could we have a single test without gauge metrics but with the multi-threading. Without that there is no signal that this code is working.

Reply:
Done